### PR TITLE
feat(alert): use support library for alert dialog

### PR DESCRIPTION
**Pull request by DomiR**
## Summary

Instead of using the AlertDialog from the android library we use the AlertDialog from the support library to render dialogs properly on Lollipop phones.

## Changelog

[Android] [Fixed] - Fix AlertDialog for older android devices by using AlertDialog from the androidx support library

## Test Plan

Visual inspection.
With old AlertDialog:
ddd1692b-2bcb-424a-9355-1b54515c6618 - Donald Duck (1)

With compat AlertDialog:
Screenshot_1973-06-21-21-23-28

### CODE REVIEW

**Comment by safaiyeh** [comment]
cc @dulmandakh

**Review by dulmandakh — APPROVED** [review]
LGTM

**Comment by DomiR** [comment]
I don't know how to fix the BUCK files... circleci doesn't seem to build properly, but I compiled it on my machine and the result can be seen above.

**Comment by DomiR** [comment]
I can't wait for the next release and I'm scared to roll with my own fork for the release 😅 

Here https://github.com/DomiR/react-native-compat-alert is reimplementation of the module, that uses androidx.appcompat.app.AlertDialog. Would be nice to see this in the next release.

File: ReactAndroid/src/main/java/com/facebook/react/modules/dialog/BUCK [hunk 1]
```diff
@@ -7,6 +7,7 @@ rn_android_library(
     labels = ["supermodule:xplat/default/public.react_native.infra"],
     provided_deps = [
         react_native_dep("third-party/android/androidx:annotation"),
+        react_native_dep("third-party/android/androidx:app"),
```
Comment (dulmandakh) [inline ReactAndroid/src/main/java/com/facebook/react/modules/dialog/BUCK:10]: try changing ** react_native_dep("third-party/android/androidx:app")** with ** react_native_dep("third-party/android/androidx:appcompat-binary")**

**Comment by dulmandakh** [comment]
Please remove everything in provided_deps, and add **react_native_dep("third-party/android/androidx:appcompat")**

**Comment by DomiR** [comment]
@dulmandakh there is still some BUCK file for the test build that needs updating... any ideas?

**Comment by dulmandakh** [comment]
You find a file that fails to compile, and usually there is a BUCK file in the same folder which might require change.

**Comment by dulmandakh** [comment]
I dug a little bit, and couldn't a fix the test failure. It seems like AppCompat components require themed context, ThemedReactContext in RN case, but RN passes ReactApplicationContext to modules constructor which is used by DialogModule to show AlertDialog. Thus tests are failing. Please help @mdvacca

**Comment by camilossantos2809** [comment]
Any updates? I have an app that is used a lot on devices with android 5/lollipop, this improvement would help me a lot. Thanks!

**Comment by DomiR** [comment]
Sucks that internal Facebook build tools are not the OS way. It builds for me, but BUCK needs to be configured differently to allow those tests to pass.

In the meantime you can use my fork, which only replaces that import: https://github.com/DomiR/react-native-compat-alert

**Comment by safaiyeh** [comment]
What versions of Android does this aim to fix?

**Comment by DomiR** [comment]
I think Lollipop and earlier... other components also use the androidx compact lib, so it's not something unusual. Without compat the Alert works on older phones with the limitations mentioned above (no vertical button stacking, overflow on small devices).

**Pull request closed without merging**